src/lib.rs

```diff
@@ -155,6 +99,14 @@ pub trait Service {
 
     /// Process the request and return the response asynchronously.
     fn call(&self, req: Self::Request) -> Self::Future;
+
+    /// Wrap this Service in a Middleware component.
+    fn wrap<M>(self, middleware: M) -> M::WrappedService where
```

Comment:
I think wrap works in both directions, in the sense of `wrap in {middleware}` and `wrap around {service}`. But I'm also fine with whatever name.